tri_mocap_common/src/mocap_tf_publisher_node.ros2.cpp

```diff
@@ -0,0 +1,114 @@
+#include <cstdlib>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <utility>
+
+// ROS
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <tf2_msgs/msg/tf_message.hpp>
+#include <tri_mocap_common/msg/mocap_state.hpp>
+
+#include "mocap_tf_publisher_node.ros2.hpp"
+
+namespace tri_mocap_common
+{
+namespace
+{
+std::string MakeTfCompatibleName(const std::string& name)
+{
+  const std::pair<char, char> ascii_numbers(0x30, 0x39);
+  const std::pair<char, char> ascii_upper_case(0x41, 0x5a);
+  const std::pair<char, char> ascii_lower_case(0x61, 0x7a);
+
+  std::ostringstream strm;
+  for (size_t idx = 0; idx < name.size(); idx++)
+  {
+    const char character = name[idx];
+    if (character >= ascii_numbers.first && character <= ascii_numbers.second)
+    {
+      strm << character;
+    }
+    else if (character >= ascii_upper_case.first
+             && character <= ascii_upper_case.second)
+    {
+      strm << character;
+    }
+    else if (character >= ascii_lower_case.first
+             && character <= ascii_lower_case.second)
+    {
+      strm << character;
+    }
+  }
+  return strm.str();
+}
+}  // namespace
+
+MocapTFPublisherNode::MocapTFPublisherNode(
+    const rclcpp::NodeOptions& options)
+    : rclcpp::Node("mocap_tf_publisher", options)
+{
+  // Default ROS params
+  const std::string DEFAULT_TF_TOPIC("/tf");
+  const std::string DEFAULT_TRACKER_TOPIC("mocap_state");
+  const bool DEFAULT_OVERRIDE_TIMESTAMPS = false;
+
+  // Get params
+  const std::string tf_topic
+      = this->declare_parameter("tf_topic", DEFAULT_TF_TOPIC);
+  const std::string tracker_topic
+      = this->declare_parameter("tracker_topic", DEFAULT_TRACKER_TOPIC);
+  override_timestamps_ = this->declare_parameter(
+      "override_timestamps", DEFAULT_OVERRIDE_TIMESTAMPS);
+
+  // Make ROS publisher + subscriber
+  tf_pub_ = this->create_publisher<TFMessage>(tf_topic, 1);
```

Comment:
For QoS I'd recommend using `tf2_ros::DynamicBroadcasterQoS()` from `tf2_ros/qos.hpp` so that this would stay compatible with any  future changes to tf's QoS settings.

Alternatively you could use `tf2_ros::TransformBroadcaster` and [`sendTransform()`](https://github.com/ros2/geometry2/blob/2687a3543bf61481b06dcf37839c7e92bed85d2e/tf2_ros/include/tf2_ros/transform_broadcaster.h#L92), but then the `tf_topic` parameter wouldn't work. The only way to change the `tf` topic would be through remapping.

Reply:
Looking at the current TF publisher QoS settings, I think we're OK keeping the smaller queue size in place for now (as I understand it, the QoS settings are otherwise the same). We can reconsider in the future if we need to.